sdk/inc/azure/core/az_http.h

```diff
@@ -184,10 +185,10 @@ AZ_NODISCARD AZ_INLINE az_result az_http_response_init(az_http_response* respons
  */
 typedef struct
 {
+  az_span reason_phrase;
```

Comment:
Can you add a comment explaining why we have these fields in this order, so folks who might want to change this later know about the rationale?

Reply:
I'll add comment, but I think optimizing for packing should be the default, and we should only comment when we make an exception from that :)

Reply:
>  I think optimizing for packing should be the default

If you feel strongly, we should first fix up the packing of all our existing structs and then I agree :)

Keeping it aligned/packed is difficult though and could be hard to correctly maintain.